content-src/components/DiscoveryStreamComponents/SafeAnchor/SafeAnchor.jsx

```diff
@@ -1,6 +1,33 @@
+import {actionCreators as ac, actionTypes as at} from "common/Actions.jsm";
 import React from "react";
 
 export class SafeAnchor extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(event) {
+    // Use dispatch instead of normal link click behavior to include referrer
+    if (this.props.dispatch) {
+      event.preventDefault();
+      const {altKey, button, ctrlKey, metaKey, shiftKey} = event;
+      this.props.dispatch(ac.OnlyToMain({
+        type: at.OPEN_LINK,
+        data: {
+          event: {altKey, button, ctrlKey, metaKey, shiftKey},
+          referrer: "https://getpocket.com/recommendations",
+          url: event.currentTarget.href,
```

Comment:
Any reason you don't want to use `props.url` here?

Reply:
SafeAnchor can clean up the url to be different (i.e., "") so we don't want to accidentally open an unsafe url via chrome. I'll add a comment

Reply:
ah, makes sense